### PR TITLE
don't keep closed cases with no dependencies in the sync log

### PR DESCRIPTION
**Pull request by czue**
fixes the latest sync 412 issue.

@snopoke @dannyroberts wait for tests

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/management/commands/sync_log_debugger.py [hunk 1]
```diff
@@ -68,5 +68,22 @@ def handle(self, *args, **options):
...
+    # utility for brute force searching for a hash
+    diff = diff or set()
+    if _get_hash(case_id_set) == expected_hash:
+        return diff
+    else:
+        if depth > 0:
+            for id in case_id_set:
+                list_to_check = case_id_set - set([id])
+                newdiff = diff | set([id])
+                result = _brute_force_search(list_to_check, expected_hash, newdiff, depth-1)
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/management/commands/sync_log_debugger.py]: E226 missing whitespace around arithmetic operator

**Comment by czue** [comment]
will look at build
```
======================================================================
ERROR: testOtherUserReassignsIndexed (casexml.apps.phone.tests.test_sync_mode.MultiUserSyncTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 45, in inner
    return helper(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 32, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/tzmigration/test_utils.py", line 28, in call_with_settings
    fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 28, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 1145, in testOtherUserReassignsIndexed
    assert_user_has_case(self, self.user, case_id, restore_id=latest_sync_log.get_id)
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 115, in assert_user_has_case
    return assert_user_has_cases(testcase, user, [case_id], return_single=True, **kwargs)
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 121, in assert_user_has_cases
    should_have=True, line_by_line=False, version=V2, **kwargs)
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 160, in check_user_has_case
    payload_string = restore_config.get_payload().as_string()
  File "corehq/ex-submodules/casexml/apps/phone/restore.py", line 467, in get_payload
    partial_response = provider.get_response(self.restore_state)
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/providers.py", line 12, in get_response
    return get_case_payload(restore_state)
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py", line 22, in get_case_payload
    return sync_op.get_payload()
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py", line 126, in get_payload
    self.restore_state.current_sync_log.prune_dependent_cases()
  File "corehq/ex-submodules/casexml/apps/phone/models.py", line 704, in prune_dependent_cases
    self.prune_case(dependent_case_id)
  File "corehq/ex-submodules/casexml/apps/phone/models.py", line 595, in prune_case
    _remove_case(to_remove)
  File "corehq/ex-submodules/casexml/apps/phone/models.py", line 575, in _remove_case
    self.case_ids_on_phone.remove(to_remove)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/jsonobject/containers.py", line 182, in remove
    raise KeyError(wrapped)
KeyError: u'other_reassigns_index_parent'
----------------------------------------------------------------------
```

**Comment by dannyroberts** [comment]
Awesome! Any sense of whether these are making a dent in the total number of 412's we're getting?

**Comment by czue** [comment]
definitely a big dent in the new sync - this bug didn't exist in the old code

**Pull request merged**